pkg/kubectl/cmd/get.go

```diff
@@ -315,5 +339,5 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 			continue
 		}
 	}
-	return utilerrors.Flatten(utilerrors.NewAggregate(allErrs))
+	return utilerrors.NewAggregate(allErrs)
```

Comment:
Ok, so looking at the latest my concern is that this is harder to understand than the previous code - all that is different is the IntoSingular(), and you could use a block at the end to check whether this was singular before returning.  I think collapsing them back down into one would be clearer.

Looking at this more,  I just noticed that you were keying on Recursive - but Recursive should be a detail inside of resource builder, and specifying recursive should automatically set singular to false (inside of the resource.Builder code).  The intent of singular was to resolve all the inputs down into "did the user give us a single object or not" and passing recursive is a definite "not".